### PR TITLE
add option to reverse buildchains

### PR DESCRIPTION
**Pull request by deads2k**
This allows a user to start with an istag and find all the builds and images that lead to it.

```bash
oc process -f examples/sample-app/application-template-stibuild.json -l build=sti | oc create -f -
[deads@deads-dev-01 origin]$ oadm build-chain istag/origin-ruby-sample --reverse
istag/origin-ruby-sample:latest
	bc/ruby-sample-build
		istag/ruby-22-centos7:latest
```

@bparees @kargakis

### CODE REVIEW

**Comment by bparees** [comment]
how does the output handle forking?

File: pkg/cmd/cli/describe/chaindescriber.go [hunk 1]
```diff
@@ -144,11 +149,46 @@ func partition(g osgraph.Graph, root graph.Node, buildInputEdgeKinds []string) o
...
+	// Filter out all but BuildConfig and ImageStreamTag nodes
+	nodeFn := osgraph.NodesOfKind(buildgraph.BuildConfigNodeKind, imagegraph.ImageStreamTagNodeKind)
+	// Filter out all but BuildInputImage and BuildOutput edges
+	edgeKinds := []string{}
+	edgeKinds = append(edgeKinds, buildInputEdgeKinds...)
+	edgeKinds = append(edgeKinds, buildedges.BuildOutputEdgeKind)
+	edgeFn := osgraph.EdgesOfKind(edgeKinds...)
+	sub := g.Subgraph(nodeFn, edgeFn)
+
+	// Filter out inbound edges to the ist of interest
```
Comment (bparees) [inline pkg/cmd/cli/describe/chaindescriber.go]: list(?)

Comment (deads2k) [inline pkg/cmd/cli/describe/chaindescriber.go]: > list(?)

ist.  imagestreamtag

Comment (bparees) [inline pkg/cmd/cli/describe/chaindescriber.go]: IST then.  (yeah it's nit picky, but it confused me, so....)

**Comment by bparees** [comment]
wouldn't mind a test case...

**Comment by deads2k** [comment]
> wouldn't mind a test case...

Sure.  When I need a break from the rebase I'll write up with some forks to deal with both comments.

**Comment by kargakis** [comment]
lgtm

**Comment by deads2k** [comment]
> how does the output handle forking?

Test added.  Output:
```
[deads@deads-dev-01 origin]$ oadm build-chain istag/child2img:latest --reverse
istag/child2img:latest
	bc/child2
		istag/parent1img:latest
			bc/parent1
				istag/ruby-22-centos7:latest
		istag/parent3img:latest
			bc/parent3
				istag/ruby-22-centos7:latest
```

**Comment by deads2k** [comment]
[merge]

**Pull request merged**